### PR TITLE
tweak repository

### LINKED ISSUE #185: add CODE_OF_CONDUCT file

### PR DESCRIPTION
**Pull request by rvhonorato**
This PR changes the repository to satisfy EOSC's SQAaaS analysis

## Changes

- Migrate from `setup.py` to `pyproject.toml` (not needed by SQAaaS tool but will make the changes easier)
- Use `poetry` to manage the dependencies in the actions / development
- Add Codacy to the CI pipeline
- Upgrade `trunk` and enable pre-commit checks
- Lint code (mostly E501)
- Added `CODE_OF_CONDUCT` which closes #185

### CODE REVIEW

**Comment by rvhonorato** [comment]
These put us on Bronze, which is as far as we can go until we make releases and publish the code;

Screenshot 2023-02-06 at 18 14 31

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,19 +2,44 @@
...
+git clone https://github.com/haddocking/arctic3d.git && cd arctic3d
+pip install .
+```
+
 ## Testing
 
 ```text
-pip install pytest
-pytest
+run pytest --cov=./ --cov-report=xml -v
```
Comment (mgiulini) [inline CONTRIBUTING.md:26]: the `run` key should be outside the text, right?

What if the user does use pip to install the package (without installing poetry)? I would leave the standard pytest line here for testing

**Review by mgiulini — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (rvhonorato) [reply CONTRIBUTING.md:26]: Oh yes, it's in the wrong place it, will fix it

Comment (rvhonorato) [reply CONTRIBUTING.md]: done, check the new version

Comment (mgiulini) [reply CONTRIBUTING.md]: now the testing runs smoothly even if the user does not use poetry, nice! I guess this can be merged :)

Comment (rvhonorato) [reply CONTRIBUTING.md]: yes! poetry is more to help us when adding dependencies and solving the conflicts, for the users it does not matter much

Comment (mgiulini) [reply CONTRIBUTING.md]: indeed!

**Pull request merged**